apps/flightsnearby/flights_nearby.star

```diff
@@ -0,0 +1,244 @@
+}
+
+# (degrees–>radians)
+def deg_to_rad(num):
+    return num * (math.pi / 180)
+
+# (radians–>degrees)
+def rad_to_deg(num):
+    return (180 * num) / math.pi
+
+def get_bounding_box(centrePoint, distance):
+    distance = int(distance)
+    if distance < 0:
+        fail("Distance must be greater than 0")
+
+    # coordinate limits
+    MIN_LAT = deg_to_rad(-90)
+    MAX_LAT = deg_to_rad(90)
+    MIN_LON = deg_to_rad(-180)
+    MAX_LON = deg_to_rad(180)
+
+    # earth's radius (km)
+    R = 6378.1
+
+    # angular distance in radians on a great circle
+    radDist = distance / R
+
+    # centre point coordinates (deg)
+    degLat = centrePoint[0]
+    degLon = centrePoint[1]
+
+    # centre point coordinates (rad)
+    radLat = deg_to_rad(degLat)
+    radLon = deg_to_rad(degLon)
+
+    # minimum and maximum latitudes for given distance
+    minLat = radLat - radDist
+    maxLat = radLat + radDist
+
+    # minimum and maximum longitudes for given distance
+    minLon = 0
+    maxLon = 0
+
+    # define deltaLon to help determine min and max longitudes
+    deltaLon = math.asin(math.sin(radDist) / math.cos(radLat))
+    if (minLat > MIN_LAT) and (maxLat < MAX_LAT):
+        minLon = radLon - deltaLon
+        maxLon = radLon + deltaLon
+        if minLon < MIN_LON:
+            minLon = minLon + 2 * math.pi
+        if maxLon > MAX_LON:
+            maxLon = maxLon - 2 * math.pi
+
+        # a pole is within the given distance
+    else:
+        minLat = math.max(minLat, MIN_LAT)
+        maxLat = math.min(maxLat, MAX_LAT)
+        minLon = MIN_LON
+        maxLon = MAX_LON
+    return [
+        str(rad_to_deg(minLat)),
+        str(rad_to_deg(minLon)),
+        str(rad_to_deg(maxLat)),
+        str(rad_to_deg(maxLon)),
+    ]
+
+def is_key_present(k):
+    if k in TAILS:
+        return TAILS[k]
+    else:
+        return TAILS["Q4"]
+
+def main(config):
+    api_key = secret.decrypt("AV6+xWcEFKIfCw67zTcYusGXoTGKrc1bOSIdg8X8UuTkEYMbEdcN67Lh1R8PxQUqFf3QxoXI39bIZz0LV8LgL11JY+/uSxEYPlrVMMEqzJMUJngnn9ZW31mmc5Hk9iNnwgfKAZ6YOadc86TTyR3Pyg78hbUfrE2brp3zYWRpsLrfJpxtIlX4UVlHtuC4qF87pQjyeLQ1wHw=")
+    location = config.get("location", DEFAULT_LOCATION)
+
+    flight_cached = cache.get("flight")
+    if flight_cached != None:
+        print("Hit! Displaying cached data.")
+        flight = json.decode(flight_cached)
+    else:
+        print("Miss! Contacting Flight Radar")
+        centrePoint = [float(location["lat"]), float(location["lng"])]
+        boundingBox = get_bounding_box(centrePoint, config.get("distance", DEFAULT_DISTANCE))
+        rep = http.get(
+            FLIGHT_RADAR_URL,
+            params = {"bl_lat": boundingBox[0], "bl_lng": boundingBox[1], "tr_lat": boundingBox[2], "tr_lng": boundingBox[3]},
+            headers = {"X-RapidAPI-Key": config.get("key", api_key), "X-RapidAPI-Host": "flight-radar1.p.rapidapi.com"},
```

Comment:
It's possible the Tidbyt mobile app will pass in an empty string when `key` is not set by the user. It would probably be a good idea to get `key` and check it against both `""` and `None` to determine if the default key should be used.